### PR TITLE
Feat/make footer configurable

### PR DESCRIPTION
**Pull request by annafatani**

### CODE REVIEW

File: src/lib/footer/Footer.svelte [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	export let endClass = '';
+	export let bottomClass = '';
+</script>
+
+<div class="max-w-screen-2xl bg-white p-4 text-sm dark:bg-gray-700 sm:p-10 2xl:mx-auto">
+	<div class={twMerge('justify-between lg:flex', className)}>
+		<div class={twMerge('flex w-full md:w-auto', startClass)}>
+			<slot name="start" />
+		</div>
+		<div class={twMerge('md:contents md:w-auto')}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:17]: twMerge not needed here

File: src/lib/footer/Footer.svelte [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			<div class={twMerge('max-lg:my-5', middleClass)}>
+				<slot name="middle" />
+			</div>
+			<div class={twMerge(endClass)}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:21]: rm `twMerge`

File: src/lib/footer/Footer.svelte [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	let className = '';
+	export { className as class };
+	export let startClass = '';
+	export let middleClass = '';
+	export let endClass = '';
+	export let bottomClass = '';
+</script>
+
+<div class="max-w-screen-2xl bg-white p-4 text-sm dark:bg-gray-700 sm:p-10 2xl:mx-auto">
+	<div class={twMerge('justify-between lg:flex', className)}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:13]: className should be applied to the outer most component

File: src/lib/footer/Footer.svelte [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	let className = '';
+	export { className as class };
+	export let startClass = '';
+	export let middleClass = '';
+	export let endClass = '';
+	export let bottomClass = '';
+</script>
+
+<div class="max-w-screen-2xl bg-white p-4 text-sm dark:bg-gray-700 sm:p-10 2xl:mx-auto">
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:12]: lets spread the `$$restProps` onto here

File: src/routes/(docs)/Footer/ExampleFooter.svelte.d.ts
Comment (jamesbirtles) [inline src/routes/(docs)/Footer/ExampleFooter.svelte.d.ts:1]: Remove this file, I'm curious why these generated types keep getting added? where are they coming from?

File: src/routes/(docs)/Footer/ExampleFooter.svelte [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+			<li><a href="#cookies" class="hover:underline">Cookies Policy</a></li>
+			<li><a href="#terms" class="hover:underline">Terms And Conditions</a></li>
+			<li><a href="#policy" class="hover:underline">Acceptable Use Policy</a></li>
+			<li><a href="#modern-slavery" class="hover:underline">Modern Slavery</a></li>
+			<li><a href="#compliments" class="hover:underline">Complaints and Compliments</a></li>
+		</ul>
+	</div>
+
+	<div slot="bottom">
+		<Typography variant="caption">All Rights Reserved | PeoplePlus &copy; 2022</Typography>
```
Comment (jamesbirtles) [inline src/routes/(docs)/Footer/ExampleFooter.svelte:52]: Might as well do 2023 😝

File: src/routes/(docs)/Footer/ExampleFooter.svelte.d.ts
Comment (annafatani) [reply src/routes/(docs)/Footer/ExampleFooter.svelte.d.ts:1]: I have no idea, maybe from testing?

Comment (annafatani) [reply src/routes/(docs)/Footer/ExampleFooter.svelte.d.ts:1]: Also maybe because I ran `npm run lint`

File: src/lib/footer/Footer.svelte [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+<div
+	class={twMerge(
+		'max-w-screen-2xl bg-white p-4 text-sm dark:bg-gray-700 sm:p-10 2xl:mx-auto',
+		className
+	)}
+	{...$$restProps}
+>
+	<div class={'justify-between lg:flex'}>
+		<div class={('flex w-full md:w-auto', startClass)}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:20]: This still needs twMerge, I didn't mean to remove it from everything, just the two instances where it wasn't actually doing anything.

File: src/lib/footer/Footer.svelte [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+			<slot name="start" />
+		</div>
+		<div class={'md:contents md:w-auto'}>
+			<div class={('max-lg:my-5', middleClass)}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:24]: this too

File: src/lib/footer/Footer.svelte [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+				<slot name="middle" />
+			</div>
+			<div class={endClass}>
+				<slot name="end" />
+			</div>
+		</div>
+	</div>
+
+	<div class={('mt-5 border-t border-gray-100 pt-5', bottomClass)}>
```
Comment (jamesbirtles) [inline src/lib/footer/Footer.svelte:33]: and here

**Review by jamesbirtles — CHANGES_REQUESTED** [review]

**Review by jamesbirtles — APPROVED** [review]

**Review by danny-people-plus — APPROVED** [review]

**Pull request merged**